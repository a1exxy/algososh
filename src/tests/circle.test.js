import {render} from "@testing-library/react";
import React from "react";
import {Circle} from "../components/ui/circle/circle";
import {ElementStates} from "../types/element-states";

describe('2: Тестирование компонента Circle (проверяем при помощи снэпшотов корректную отрисовку элемента)', () => {
  it('2.1: без буквы', () => {
    const {container} = render(<Circle/>)
    expect(container).toMatchSnapshot()
  })
  it('2.2: с буквами', () => {
    const testData = "aaa"
    const {container} = render(<Circle letter={testData}/>)
    expect(container.querySelector("p.letter").textContent).toBe(testData)
    expect(container).toMatchSnapshot()
  })
  it('2.3: с head', () => {
    const testData = "aaa"
    const {container} = render(<Circle head={testData}/>)
    expect(container.querySelector("div.head").textContent).toBe(testData)
    expect(container).toMatchSnapshot()
  })
  it('2.4: с react-элементом в head', () => {
    const testData = "aaa"
    const {container} = render(<Circle head={<Circle letter={testData}/>}/>)
    expect(container.querySelector("div.head p.letter").textContent).toBe(testData)
    expect(container).toMatchSnapshot()
  })
  it('2.5: с tail', () => {
    const testData = "aaa"
    const {container} = render(<Circle tail={testData}/>)
    expect(container.querySelector("div.tail30").textContent).toBe(testData)
    expect(container).toMatchSnapshot()
  })
  it('2.6: с react-элементом в tail', () => {
    const testData = "aaa"
    const {container} = render(<Circle tail={<Circle letter={testData}/>}/>)
    expect(container.querySelector("div.tail30 p.letter").textContent).toBe(testData)
    expect(container).toMatchSnapshot()
  })
  it('2.7: с index', () => {
    const testData = "1"
    const {container} = render(<Circle index={testData}/>)
    expect(container.querySelector("p.index").textContent).toBe(testData)
    expect(container).toMatchSnapshot()
  })
  it('2.8: с пропом isSmall ===  true', () => {
    const {container} = render(<Circle isSmall={true}/>)
    expect(container).toMatchSnapshot()
  })
  it('2.9: в состоянии default', () => {
    const {container} = render(<Circle state={ElementStates.Default}/>)
    expect(container).toMatchSnapshot()
  })
  it('2.10: в состоянии changing', () => {
    const {container} = render(<Circle state={ElementStates.Changing}/>)
    expect(container).toMatchSnapshot()
  })
  it('2.11: в состоянии modified', () => {
    const {container} = render(<Circle state={ElementStates.Modified}/>)
    expect(container).toMatchSnapshot()
  })
})