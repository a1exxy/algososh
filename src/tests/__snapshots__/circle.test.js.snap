// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2: Тестирование компонента Circle (проверяем при помощи снэпшотов корректную отрисовку элемента) 2.1: без буквы 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`2: Тестирование компонента Circle (проверяем при помощи снэпшотов корректную отрисовку элемента) 2.2: с буквами 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        aaa
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`2: Тестирование компонента Circle (проверяем при помощи снэпшотов корректную отрисовку элемента) 2.3: с head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      aaa
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`2: Тестирование компонента Circle (проверяем при помощи снэпшотов корректную отрисовку элемента) 2.4: с react-элементом в head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle   default"
        >
          <p
            class="text text_type_circle text_color_input letter"
          >
            aaa
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`2: Тестирование компонента Circle (проверяем при помощи снэпшотов корректную отрисовку элемента) 2.5: с tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
    >
      aaa
    </div>
  </div>
</div>
`;

exports[`2: Тестирование компонента Circle (проверяем при помощи снэпшотов корректную отрисовку элемента) 2.6: с react-элементом в tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle   default"
        >
          <p
            class="text text_type_circle text_color_input letter"
          >
            aaa
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`2: Тестирование компонента Circle (проверяем при помощи снэпшотов корректную отрисовку элемента) 2.7: с index 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      1
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>
</div>
`;

exports[`2: Тестирование компонента Circle (проверяем при помощи снэпшотов корректную отрисовку элемента) 2.8: с пропом isSmall ===  true 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle  small default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`2: Тестирование компонента Circle (проверяем при помощи снэпшотов корректную отрисовку элемента) 2.9: в состоянии default 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`2: Тестирование компонента Circle (проверяем при помощи снэпшотов корректную отрисовку элемента) 2.10: в состоянии changing 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   changing"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`2: Тестирование компонента Circle (проверяем при помощи снэпшотов корректную отрисовку элемента) 2.11: в состоянии modified 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   modified"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;
